pkg/cincinnati/cincinnati.go

```diff
@@ -34,19 +34,36 @@ func NewClient(id uuid.UUID, proxyURL *url.URL, tlsConfig *tls.Config) Client {
 // Update is a single node from the update graph.
 type Update node
 
+// Error is returned when are unable to get updates.
+type Error struct {
+	// Reason is the reason suggested for the ClusterOperator status condition.
+	Reason  string
+
+	// Message is the message suggested for the ClusterOperator status condition.
+	Message string
+
+	// cause is the upstream error, if any, being wrapped by this error.
+	cause error
+}
+
+// Error serializes the error as a string, to satisfy the error interface.
+func (err *Error) Error() string {
+	return fmt.Sprintf("%s: %s", err.Reason, err.Message)
+}
+
 // GetUpdates fetches the next-applicable update payloads from the specified
 // upstream Cincinnati stack given the current version and channel. The next-
 // applicable updates are determined by downloading the update graph, finding
 // the current version within that graph (typically the root node), and then
 // finding all of the children. These children are the available updates for
 // the current version and their payloads indicate from where the actual update
 // image can be downloaded.
-func (c Client) GetUpdates(upstream string, arch string, channel string, version semver.Version) ([]Update, error) {
+func (c Client) GetUpdates(upstream string, arch string, channel string, version semver.Version) ([]Update, *Error) {
 	transport := http.Transport{}
 	// Prepare parametrized cincinnati query.
 	cincinnatiURL, err := url.Parse(upstream)
 	if err != nil {
-		return nil, fmt.Errorf("failed to parse upstream URL: %s", err)
+		return nil, &Error{Reason: "InvalidURI", Message: fmt.Sprintf("failed to parse upstream URL: %s", err)}
```

Comment:
empty upstream will create cryptic error message as it's not quoted.

Reply:
Empty upstream is handled by an earlier guard.  But I can shift the parsing over there to make that more clear.

Reply:
> Empty upstream is handled by an earlier guard.

hmm, that not really the best way to handle this.. making sure the error msg are clear from this function is still important imo.

Reply:
Shifted to live next to the existing `len` guard with bff5982.  Or were you asking me to shift both of those guards down into `GetUpdates` instead of having them live up in `calculateAvailableUpdatesStatus`?